test/mix/tasks/kitto.new_test.exs

```diff
@@ -0,0 +1,63 @@
+Code.require_file "../../../installer/lib/kitto_new.ex", __DIR__
+defmodule Mix.Tasks.Kitto.NewTest do
+  use ExUnit.Case, async: false
+  import Plug.Test
+  import Kitto.FileAssertionHelper
+  import ExUnit.CaptureIO
+
+  setup do
+    Mix.Task.clear
+    # The shell asks to install npm and mix deps.
+    # We will politely say not.
+    send self(), {:mix_shell_input, :yes?, false}
+    :ok
+  end
+
+  test "returns the current verions when `--version` is provided" do
+    Mix.Tasks.Kitto.New.run(["--version"])
+    kitto_version = "Kitto v#{Mix.Project.config[:version]}"
+    assert_received {:mix_shell, :info, [^kitto_version] }
+  end
+
+  test "fails when invalid application name is provided" do
+    assert_raise Mix.Error, fn ->
+      Mix.Tasks.Kitto.New.run(["@1@#%$#@"])
+    end
+  end
+
+  test "fails when only provindg a switch" do
+    assert_raise Mix.Error, fn ->
+      Mix.Tasks.Kitto.New.run(["-b"])
+    end
+  end
+
+  test "bootstraps generated project" do
```

Comment:
I think this test can be split in:

1. case asserting that files are actually generated in the project directory
1. case asserting that the generated project works (a request for a dashboard responds with 200 OK)